grpc-stubs/aio.pyi

```diff
@@ -0,0 +1,258 @@
+) -> Channel: ...
+
+"""Create Server"""
+
+def server(
+    migration_thread_pool: typing.Optional[futures.Executor] = None,
+    handlers: typing.Optional[typing.Sequence[GenericRpcHandler]] = None,
+    interceptors: typing.Optional[typing.Sequence[typing.Any]] = None,
+    options: typing.Optional[_Options] = None,
+    maximum_concurrent_rpcs: typing.Optional[int] = None,
+    compression: typing.Optional[Compression] = None,
+) -> Server: ...
+
+"""Channel Object"""
+
+RequestSerializer = typing.Callable[[typing.Any], bytes]
+ResponseDeserializer = typing.Callable[[bytes], typing.Any]
+
+class Channel:
+    async def close(self, grace: typing.Optional[float]) -> None: ...
+    def stream_stream(
+        self,
+        method: str,
+        request_serializer: typing.Optional[RequestSerializer],
+        response_deserializer: typing.Optional[ResponseDeserializer],
+    ) -> StreamStreamMultiCallable: ...
+    def stream_unary(
+        self,
+        method: str,
+        request_serializer: typing.Optional[RequestSerializer],
+        response_deserializer: typing.Optional[ResponseDeserializer],
+    ) -> StreamUnaryMultiCallable: ...
+    def unary_stream(
+        self,
+        method: str,
+        request_serializer: typing.Optional[RequestSerializer],
+        response_deserializer: typing.Optional[ResponseDeserializer],
+    ) -> UnaryStreamMultiCallable: ...
+    def unary_unary(
+        self,
+        method: str,
+        request_serializer: typing.Optional[RequestSerializer],
+        response_deserializer: typing.Optional[ResponseDeserializer],
+    ) -> UnaryUnaryMultiCallable: ...
+    async def __aenter__(self) -> Channel: ...
+    async def __aexit__(
+        self,
+        exc_type: typing.Optional[typing.Type[BaseException]],
+        exc_val: typing.Optional[BaseException],
+        exc_tb: typing.Optional[TracebackType],
+    ) -> typing.Optional[bool]: ...
+    async def channel_ready(self) -> None: ...
+
+"""Server Object"""
+
+class Server:
+    def add_generic_rpc_handlers(
+        self,
+        generic_rpc_handlers: typing.Iterable[GenericRpcHandler],
+    ) -> None: ...
+
+    # Returns an integer port on which server will accept RPC requests.
+    def add_insecure_port(self, address: str) -> int: ...
+
+    # Returns an integer port on which server will accept RPC requests.
+    def add_secure_port(
+        self, address: str, server_credentials: ServerCredentials
+    ) -> int: ...
+    async def start(self) -> None: ...
+
+    # Grace period is in seconds.
+    async def stop(self, grace: typing.Optional[float] = None) -> None: ...
+
+    # Returns a bool indicates if the operation times out. Timeout is in seconds.
+    async def wait_for_termination(
+        self, timeout: typing.Optional[float] = None
+    ) -> bool: ...
+
+"""Client-Side Context"""
+
+DoneCallbackType = typing.Callable[[typing.Any], None]
+# FIXME
+EOFType = typing.Any
```

Comment:
I see another FIXME here. I traced this one back to [here](https://github.com/grpc/grpc/blob/master/src/python/grpcio/grpc/aio/_typing.py#L32) and [here](https://github.com/grpc/grpc/blob/master/src/python/grpcio/grpc/aio/_typing.py#L19); this seems like it could be tricky to get a decent type for?

Reply:
Yep, this seems like it would be really hard to type. Any avoids type errors though, so it seems a good first approximation here.

Reply:
AIUI, a better opaque type here might be `object` (unless there's something better that has turned up recently). Would you please try `object` instead of `typing.Any` and add a typing test?